### ISSUE TITLE
Document lack of default argument support?

### ISSUE THREAD

**Issue by DavisVaughan** [r-lib/cpp11#162]
I noticed in https://github.com/r-lib/cpp11/issues/4 that default arguments are not supported. This is fine, but it might be worth documenting somewhere in the Rcpp converting vignette.

lubridate has:

```cpp
Rcpp::newDatetimeVector C_update_dt(const Rcpp::NumericVector& dt,
                                    const Rcpp::IntegerVector& year,
                                    const Rcpp::IntegerVector& month,
                                    const Rcpp::IntegerVector& yday,
                                    const Rcpp::IntegerVector& mday,
                                    const Rcpp::IntegerVector& wday,
                                    const Rcpp::IntegerVector& hour,
                                    const Rcpp::IntegerVector& minute,
                                    const Rcpp::NumericVector& second,
                                    const SEXP tz = R_NilValue,
                                    const bool roll = false,
                                    const int week_start = 7) {
```

Translating to:

```r
C_update_dt <- function(dt, year, month, yday, mday, wday, hour, minute, second, tz = NULL, roll = FALSE, week_start = 7L) {
  .Call(`_lubridate_C_update_dt`, dt, year, month, yday, mday, wday, hour, minute, second, tz, roll, week_start)
}
```

If I recompile this with cpp11 to start the process of moving things over, it still compiles but the default arguments disappear. I only learned that there were no longer default arguments when I tried running the tests and ran into failures.

**Comment by vspinu** [comment]
I would very much like having the support instead. Now I have to manage extra wrapper around cpp11 wrapper which with many arguments for interactive workflows quickly becomes tedious. With Rcpp, most of the time one can build self-contained functions and forget about the R level wrapper. Convenient.

**Comment by jimhester** [comment]
While I do understand the use case you mention @vspinu, in practice I have found for the vast majority of functions I end up having an R wrapper anyway, as generally there are things like argument checking that are less awkward to do with R code vs C++.

In addition in order to support C++ only function definitions you would also need to support roxygen comments for C++ code as Rcpp does.

We decided implementing two moderately complex features to enable a workflow that isn't actually used that often in practice was not a worthwhile tradeoff at this time.

Perhaps in the future that calculation may change.
